src/test.c

```diff
@@ -145,16 +155,20 @@ void test_long_simulation_history() {
   for (i = 0; i < length; i++) {
     sample = q->event(q, &dt);
     values[i] = sample;
-    insert_reaction(sh, sample);
+    times[i] = dt;
+    insert(sh, sample, dt);
   }
 
   i = 0;
   bool flag = true;
   Chunk *chunk = sh->first_chunk;
   while (chunk) {
     for (j = 0; j < chunk->next_free_index; j++) {
-      if (chunk->data[j] != values[i])
+      if (chunk->data[j].reaction != values[i]) {
```

Comment:
I would do ` if ((chunk->data[j].reaction != values[i]) || (chunk->data[j].time != times[i]))` here instead of the else if

Reply:
I was going for shorter lines, but if that's not a priority, sure.